### ISSUE TITLE
Error 'url-not-found' even though image exists

### ISSUE THREAD

**Issue by suve** [hughsie/appstream-glib#200]
I'm currently packaging [cool-retro-term](https://github.com/Swordfish90/cool-retro-term) and I've encountered an issue with `appstream-util validate-relax`. I'm using the appdata file found in the repo, with minor changes (I removed the `width=""` and `height=""` from `<image>` since they do not match).

Anyways, the screenshots are defined as follows:
```
  <screenshots>
    <screenshot type="default">
      <caption>Default amber look</caption>
      <image>https://camo.githubusercontent.com/2443e662e95733ba6ae331f391f6ec036d1ee7fd/687474703a2f2f692e696d6775722e636f6d2f4e5566766e6c752e706e67</image>
    </screenshot>
    <screenshot>
      <caption>Apple II look</caption>
      <image>https://camo.githubusercontent.com/44a19842d532555c7b02bf6b4b4684add9edf18c/687474703a2f2f692e696d6775722e636f6d2f4d4d6d4d3648742e706e67</image>
    </screenshot>
  </screenshots>
```

Even though the images can be viewed in a browser, or fetched with `wget` or `curl` just fine, `appstream-util validate-relax` says:
```
FAILED:
? url-not-found         : <screenshot> url not found [https://camo.githubusercontent.com/44a19842d532555c7b02bf6b4b4684add9edf18c/687474703a2f2f692e696d6775722e636f6d2f4d4d6d4d3648742e706e67]
? url-not-found         : <screenshot> url not found [https://camo.githubusercontent.com/2443e662e95733ba6ae331f391f6ec036d1ee7fd/687474703a2f2f692e696d6775722e636f6d2f4e5566766e6c752e706e67]
```
One thing to note is that the file passed validation using version `0.6.13` of `libappstream-glib`, so the bug must have been introduced later.

**Comment by suve** [comment]
It just occurred to me that the packaging guidelines in Fedora say to use the `--nonet` flag, so I guess it's actually a matter of the package builders being limited to their LAN (which makes sense, as a way to prevent downloading stuff during the package build). So this is most likely a false flag. Sorry for that.

Still, I wonder - maybe it'd make sense to change the error code to something like `url-not-reachable`? Personally, `url-not-found` reminds me of HTTP 404, which is an entirely different situation from "cannot reach server".

**Comment by hughsie** [comment]
Ahh, so you're building in something like COPR? In which case you're right, there is no networking and you have to use --nonet or ignore the validation warning. You're right about returning a better warning, do you want to have a go at a patch? We'd really appreciate it. It should just be a case of adding handling the `soup_session_send_message` return code more cleverly in `libappstream-glib/as-app-validate.c` specifically `ai_app_validate_image_check()`.

**Comment by suve** [comment]
Sure, I'll take a look at the code and try to come up with something. Thanks for the tips.

**Comment by suve** [comment]
Personally I consider this issue to be solved by PR #214, so I'm taking the liberty and closing this thread.
